### ISSUE TITLE
stan_sundials_printf_override.hpp is not that helpful

### ISSUE THREAD

**Issue by bgoodri** [stan-dev/math#1008]
## Description

The stan_sundials_printf_override.hpp file is intended to make sundials print output compatible with the interface, but it does not actually do that. First, it uses variadic macros, which are legal under C++11 but sundials seems not to compile with a C++ compiler anymore. You can compile sundials with a C compiler, but variadic macros are not part of the C standard used by `gcc-4.9` on Windows. There is a standard approach, which is to use `__VA_ARGS__`, but stan_sundials_printf_override.hpp just spits stuff onto stdout, which is what we are trying to avoid. To make this all work with R, I have to change stan_sundials_printf_override.hpp to
```
#ifndef STAN_MATH_SUNDIALS_PRINTF_OVERRIDE_HPP
#define STAN_MATH_SUNDIALS_PRINTF_OVERRIDE_HPP

#include <R_ext/Print.h>

#define STAN_SUNDIALS_PRINTF(...)   Rprintf(__VA_ARGS__)
#define STAN_SUNDIALS_FPRINTF(...)

#endif
```

## Example

Replace stan_sundials_printf_override.hpp in StanHeaders with stan_sundials_printf_override.hpp from Math and try to install the StanHeaders package on Windows with `gcc-4.9`. You get a compiler error.

## Expected Output

None

## Additional Information

None

## Current Math Version
v2.18.0

**Comment by rok-cesnovar** [comment]
Has this issue been observed since #1098 was merged. If not, please close this issue.

**Comment by mcol** [comment]
I believe this can be closed, as StanHeaders 2.19.0 includes exactly the same `stan_sundials_printf_override.hpp` file as was introduced by #1098, so no other workaround is necessary.

@bgoodri please reopen if my understanding is incorrect.
